### PR TITLE
Emit generic events from tx wasms

### PR DESCRIPTION
**Pull request by sug0**
## Describe your changes

Emit core events (i.e. `namada_core::event::Event`) from tx wasms. To this end, we leverage IBC's existing wasm infrastructure.

~~**NOTE:** IBC e2e tests are expected to fail, just like in the base PR.~~

## Indicate on which release or other PRs this topic is based on

#3032

Diff for review: https://github.com/anoma/namada/compare/tiago/generalize-events..tiago/emit-generic-events-wasm
(https://github.com/anoma/namada/pull/3088/files/2fc3cc8bbb6302c1bbb3ccdee26d271ed6bae092..0ac23e8a2d514d30a94abb42eafc8c0d6f5ccebd)

## Checklist before merging to `draft`
- [x] I have added a changelog
- [x] Git history is in acceptable state

### CODE REVIEW

File: crates/apps/src/lib/node/ledger/shell/finalize_block.rs [hunk 1]
```diff
@@ -380,29 +383,16 @@ where
...
                         }
                         // events from other sources
-                        response.events.extend(
-                            // ibc events
-                            result
-                                .ibc_events
-                                .iter()
-                                .cloned()
-                                .map(|ibc_event| {
-                                    ibc_event.with(Height(height)).into()
```
Comment (yito88) [inline crates/apps/src/lib/node/ledger/shell/finalize_block.rs:390]: Hermes requires the height in IBC events. We have to add `height` in another part if removing here.

Comment (sug0) [reply crates/apps/src/lib/node/ledger/shell/finalize_block.rs:390]: we do add it yeah

**Review by tzemanovic — APPROVED** [review]
changes look great! Will the e2e tests work after hermes update like in the base?

**Comment by sug0** [comment]
> changes look great! Will the e2e tests work after hermes update like in the base?

haven't tested, but I think nothing broke between the base and the tip of this branch. I'll make sure to run them again tho, manually

**Review by batconjurer — DISMISSED** [review]

**Review by yito88 — APPROVED** [review]

**Pull request merged**